syntax = "proto3";
package sports_keypoints_proto;

service UserService {
    rpc CreateUser(CreateUserRequest) returns (CreateUserResponse) {}
    rpc RegisterUser(RegisterUserRequest) returns (RegisterUserResponse) {}
    rpc ReadUser(ReadUserRequest) returns (User) {}
    rpc UpdateUser(UpdateUserRequest) returns (User) {}
    rpc DeleteUser(DeleteUserRequest) returns (DeleteUserResponse) {}
}

message CreateUserRequest {
    string user_name = 1;
    string password = 2;
    string email = 3;
}

message CreateUserResponse {
    bool success = 1;
}

message RegisterUserRequest {
    string user_name = 1;
    string password = 2;
}

message RegisterUserResponse {
    bool success = 1;
    string session_token = 2;
}

message ReadUserRequest {
    string session_token = 1;
}

message ReadUserResponse {
    bool success = 1;
    User user = 2;
}

message UpdateUserRequest {
    string session_token = 1;
    string user_name = 2;
    string password = 3;
    string email = 4;
}

message UpdateUserResponse {
    bool success = 1;
    User updated_user = 3;
}

message DeleteUserRequest {
    string session_token = 1;
}

message DeleteUserResponse {
    bool success = 1;
}

message User {
    string user_name = 1;
    string email = 2;
}
